pennylane/qinfo/transforms.py

```diff
@@ -14,26 +14,28 @@
 """QNode transforms for the quantum information quantities."""
 # pylint: disable=import-outside-toplevel, not-callable
 import functools
-from typing import Sequence, Callable
+from typing import Callable, Sequence
 
 import pennylane as qml
-from pennylane.devices import DefaultQubit
-from pennylane.measurements import StateMP
+from pennylane.devices import DefaultQubit, DefaultMixed
+from pennylane.measurements import StateMP, DensityMatrixMP
+from pennylane.tape import QuantumTape
 from pennylane.transforms import adjoint_metric_tensor, metric_tensor
 from pennylane.transforms.core import transform
 
 
-def reduced_dm(qnode, wires):
+@transform
+def reduced_dm(tape: QuantumTape, wires, **kwargs) -> (Sequence[QuantumTape], Callable):
     """Compute the reduced density matrix from a :class:`~.QNode` returning
     :func:`~pennylane.state`.
 
     Args:
-        qnode (QNode): A :class:`~.QNode` returning :func:`~pennylane.state`.
+        tape (QuantumTape): A :class:`~.QuantumTape` returning :func:`~pennylane.state`.
         wires (Sequence(int)): List of wires in the considered subsystem.
 
     Returns:
-        func: Function which wraps the QNode and accepts the same arguments. When called,
-        this function will return the density matrix.
+        tuple(Sequence[QuantumTape], Callable): Sequence of transformed tapes and a
```

Comment:
Here we have to add the case of qfunc and qnode because we have the decorator. Could you check the signature added in the qfunc PR?